### DISCUSSION TITLE
FastAPI-Users v10.1.2 issue with cookie authentication

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#1047] [Q&A]
It seems that #1037 breaks cookie authentication in [jupyverse](https://github.com/jupyter-server/jupyverse).
Not sure what's going on, do you have any clue?

**Comment by frankie567** [comment]
Oh OK, I think I understand what's going on. Change this part:

```py
class GitHubTransport(CookieTransport):
    async def get_login_response(self, token: str, response: Response):
        await super().get_login_response(token, response)
        response.status_code = status.HTTP_302_FOUND
        response.headers["Location"] = "/lab"
```

to this:

```py
class GitHubTransport(CookieTransport):
    async def get_login_response(self, token: str, response: Response):
        response = await super().get_login_response(token, response)
        response.status_code = status.HTTP_302_FOUND
        response.headers["Location"] = "/lab"
        return response
```

It's because we changed how we generate the response in the base class. I think I've to harmonize the approach here so the base class always return a response instead of working with the "virtual" response of FastAPI; it'll be less ambiguous.

**Comment by davidbrochart** [comment]
> Do you have any stack trace / error log to help us spot the issue?

There is the CI [using v10.1.3](https://github.com/jupyter-server/jupyverse/runs/7487908603?check_suite_focus=true), and [using v10.1.1](https://github.com/jupyter-server/jupyverse/runs/7216657213?check_suite_focus=true).

Thanks, I tried your change but it doesn't help in the case of the cookie transport:
```py
cookie_transport = CookieTransport(cookie_secure=get_auth_config().cookie_secure)
```

**Reply by frankie567** [reply]
Ok. I'll rollback the change.

**Reply by davidbrochart** [reply]
:+1:

**Reply by frankie567** [reply]
Tracked in #1048
